src/bika/lims/api/__init__.py

```diff
@@ -132,41 +131,35 @@ def create(container, portal_type, *args, **kwargs):
     :returns: The new created object
     """
     from bika.lims.utils import tmpID
-    if kwargs.get("title") is None:
-        kwargs["title"] = "New {}".format(portal_type)
 
-    # generate a temporary ID
-    tmp_id = tmpID()
+    id = kwargs.pop("id", tmpID())
+    title = kwargs.pop("title", "New {}".format(portal_type))
 
     # get the fti
     types_tool = get_tool("portal_types")
     fti = types_tool.getTypeInfo(portal_type)
 
     if fti.product:
-        obj = _createObjectByType(portal_type, container, tmp_id)
+        obj = _createObjectByType(portal_type, container, id)
+        obj.processForm()
+        obj.edit(title=title, **kwargs)
```

Comment:
Wouldn't be necessary to notify the modification explicitly?

```python
notify(ObjectModifiedEvent(obj))
```
I recall there are some subscribers for `ObjectModifiedEvent` (for `Contact`, `Batch`, ...)
Probably for both ATs and DX.

Also, what if we do a `noLongerProvides(IAuditLog, obj)` before the edit and an `alsoProvides(IAuditLog, obj)` afterwards?

Reply:
In my opinion this modification event was always wrong in that function, because we are creating an object and not modifying it. Therefore, I think it is better to handle that internally as just one transaction.

Reply:
Understand, however, we'll need to keep this in mind. I think `Batch` type is the only one that might be affected by this change in core, but not sure atm about the scope of this change in other add-ons (e.g. jsonapi)